File: codeforces/1800~1899/1859/c.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;

  vector<int> p(n);
  iota(p.begin(), p.end(), 1);
  
  i64 ans = 0;
  for (int i = n-1; i >= 0; --i) {
    vector<int> p2 = p;
    reverse(p2.begin() + i, p2.end());

    i64 cost = 0;
    i64 max_val = 0;
    for (int j = 0; j < n; ++j) {
      max_val = max(max_val, (i64)(j+1) * p2[j]);
      cost += (j+1) * p2[j];
    }
    cost -= max_val;
    ans = max(ans, cost);
  }
  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1859/c.rs
use std::io;
use std::cmp::max;


fn read_int() -> i32 {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    return s.trim().parse::<i32>().expect("invalid input");
}

fn solve() {
    let n = read_int() as usize;
    let mut p = Vec::new();
    for x in 1..(n+1) {
        p.push(x);
    }

    let mut ans: i64 = 0;
    for i in (0..n).rev() {
        let mut p2 = p.clone();
        p2[i..n].reverse();
        
        let mut cost: i64 = 0;
        let mut max_val: i64 = 0;
        for j in 0..n {
            let c = (j+1) as i64 * p2[j] as i64;
            max_val = max(max_val, c);
            cost += c;
        }
        cost -= max_val;
        ans = max(ans, cost);
    }

    println!("{}", ans);
}


fn main() {
    let t = read_int();
    for _t in 0..t {
        solve();
    }
}
